components/builder-api/src/server/models/origins.rs

```diff
@@ -0,0 +1,43 @@
+use super::db_id_format;
+use chrono::NaiveDateTime;
+use diesel;
+use diesel::pg::PgConnection;
+use diesel::result::QueryResult;
+use diesel::sql_types::BigInt;
+use diesel::RunQueryDsl;
+use server::schema::origin::*;
+
+#[derive(Debug, Serialize, Deserialize, QueryableByName)]
+#[table_name = "origins"]
+pub struct Origin {
+    #[serde(with = "db_id_format")]
+    pub id: i64,
+    #[serde(with = "db_id_format")]
+    pub owner_id: i64,
+    pub name: String,
+    pub created_at: Option<NaiveDateTime>,
+    pub updated_at: Option<NaiveDateTime>,
+    pub default_package_visibility: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, QueryableByName)]
+#[table_name = "origins_with_stats"]
+pub struct OriginWithStats {
```

Comment:
I'm not sure I like the pattern of creating a different OriginWithStats struct... can we not modify the Origin struct?

Reply:
The struct has to match the return type of the function unfortunately. I think we could fix this if we used regular diesel functions for querying and joining but by using stored procedures the compiler tries to type-check based on the returned "table" data. When we return origin rows in the future without a package count, we'll need the regular origin struct. I made an attempt at a struct like: 

```
pub struct OriginsWithStats {
  pub origin: Origin,
  pub package_count: i64
} 
```
but I couldn't get the return types right from the sql function.